Automattic/jetpack
# Masterbar: Async the admin menu upsell nudge

Currently, the admin menu upsell nudge in the admin menu can take between 200 and 300 milliseconds (and it takes around a second on one of my atomic sites), and that's quite a solid portion of the admin TTFB, and the editor in particular.

This diff is an experiment to make an async version of it. Instead of querying JITM on page load, we do it asynchronously after page load.

It only makes sense for this to be async, because it's non-critical functionality.

This is part of the effort to improve the performance of the editor.

See D79729-code for a working implementation of this for WP.com simple sites.

Fixes #23519

#### Changes proposed in this Pull Request:
* Masterbar: Async the admin menu upsell nudge

#### Jetpack product discussion
No related product discussion.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* Get an Atomic site that currently displays a nudge below the site card in the sidebar:
![](https://cldup.com/RM34cdfp7V.png)
* Install Jetpack Beta on that site.
* Use Jetpack Beta to apply this branch on your site.
* Verify that on the following pages, the nudge loads correctly and appears well:
  * `/wp-admin/post-new.php`
  * `/wp-admin/post.php?....` - the page to edit an existing post
  * `/wp-admin/edit.php`
* Verify that dismissing the nudge works well. 
* Apply D79954-code on your WP.com sandbox.
* Sandbox a WP.com simple site that currently displays a nudge below the site card in the sidebar.
* Verify that on the following pages, the nudge loads correctly and appears well:
  * `/wp-admin/post-new.php`
  * `/wp-admin/post.php?....` - the page to edit an existing post
  * `/wp-admin/edit.php`
* Verify that dismissing the nudge works well.
